microsoft-edge/extensions-chromium/developer-guide/sidebar.md

```diff
@@ -0,0 +1,625 @@
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#examples -->
+
+For <!--more--> Sidebar API extensions demos, explore any of the following extensions at Chrome docs:
+
+* [Site-specific side panel example](https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/functional-samples/cookbook.sidepanel-site-specific)
+* [Multiple side panels example](https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/functional-samples/cookbook.sidepanel-multiple)
+* [Global side panel example](https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/functional-samples/cookbook.sidepanel-global)
+* [Dictionary side panel example](https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/functional-samples/sample.sidepanel-dictionary)
+
+
+<!-- ====================================================================== -->
+## Types for parameters of sidebar methods
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#type (empty heading) -->
+
+The sidebar methods take parameters that are of the following types, or return a value that's one of the following types.
+
+* [GetPanelOptions](#getpaneloptions)
+* [PanelBehavior](#panelbehavior)
+* [PanelOptions](#paneloptions)
+* [SidePanel](#sidepanel)
+
+
+<!-- ====================================================================== -->
+#### GetPanelOptions
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#type-GetPanelOptions -->
+
+Get panel options, including `tabId`.
+
+<!-- ---------- -->
+**PROPERTIES**
+
+<!-- ---------- -->
+###### tabId
+
+Type: `number`, optional
+
+If `tabId` is specified, the sidebar options for the given tab are returned (by the method that returns this type).  If `tabId` isn't specified, returns the default sidebar options (used for any tab that doesn't have specified settings).
+
+
+<!-- ====================================================================== -->
+#### PanelBehavior
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#type-PanelBehavior -->
+
+Panel behavior, including `openPanelOnActionClick`.
+
+<!-- ---------- -->
+**PROPERTIES**
+
+###### openPanelOnActionClick
+
+Whether clicking the extension's icon toggles showing the extension's entry in the sidebar.  The default is `false`.
+
+Type: `boolean`, optional
+
+
+<!-- ====================================================================== -->
+#### PanelOptions
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#type-PanelOptions -->
+
+Panel options, including `enabled`, `path`, and `tabId`.
+
+<!-- ---------- -->
+**PROPERTIES**
+
+###### enabled
+
+Whether the sidebar is enabled.  The default is `true`.
+
+Type: `boolean`, optional
+
+
+###### path
+
+The path to the sidebar HTML file to use.  This must be a local resource within the extension package.
+
+Type: `string`, optional
+
+The default is a blank page with no UI.
```

Comment:
This is not true - why do we have this sentence? It's not on the Chrome documentation.

Reply:
todo: if optional, state when happens if not specified